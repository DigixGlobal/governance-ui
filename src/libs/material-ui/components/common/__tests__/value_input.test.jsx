import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import ValueInput from '../value_input';

jest.mock('../../../helpers/stringUtils');

const formData = {
  testForm: 'test',
};

const sendAll = {
  toNumber: () => 1,
};

describe('<ValueInput />', () => {
  test('renders correctly', () => {
    const component = shallow(
      <ValueInput
        color="red"
        formChange={jest.fn}
        formData={formData}
        name="testForm"
        symbol="GBP"
        sendAll={sendAll}
      />,
    );

    expect(toJson(component)).toMatchSnapshot();
  });
});
