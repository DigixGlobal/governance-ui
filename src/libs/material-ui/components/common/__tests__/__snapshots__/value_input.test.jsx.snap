// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ValueInput /> renders correctly 1`] = `
<FormField>
  <Input
    action={true}
    className="labeled"
    fluid={true}
    name="testForm"
    onChange={[Function]}
    placeholder="e.g. \`5\`"
    step="any"
    type="number"
    value="test"
  >
    <Label
      color="red"
      content="GBP"
    />
    <input />
    <Button
      as="button"
      basic={true}
      content="Send All undefined"
      icon="plus"
      onClick={[Function]}
    />
  </Input>
  <Connect(CryptoPrice)
    amount={NaN}
    symbol="GBP"
  />
</FormField>
`;
